cmd/kube-scheduler/app/server.go

```diff
@@ -58,8 +58,11 @@ import (
 	"k8s.io/klog"
 )
 
-// NewSchedulerCommand creates a *cobra.Command object with default parameters
-func NewSchedulerCommand() *cobra.Command {
+// Option configures a framework.Registry.
+type Option func(framework.Registry) error
```

Comment:
`Option` is not a great name for a function. It should start with a verb or has an action in the name. Something like `pluginAdder`.

Reply:
we are not naming a function, we are naming a type. It's a fairly well accepted pattern in golang at this point. https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis

Reply:
alright. I guess I should accept it since it is an accepted pattern. :slightly_smiling_face:

Reply:
I wish "an accepted pattern" wasn't the reason you approved. It's always good to question whether established patterns are serving us. In this specific case, I think it makes more sense from outside the package. If you would read the godoc, you would see,

```go
func New(a, b int, x, y string, opts... Option) {/*...*/}

func WithFooCount(i int) Option {/*...*/}
func WithBarName(name string) Option {/*...*/}
```

If you have that API, it actually doesn't matter to me what the `Option` type looks like. It just so happens that in golang, the func-based implementation is pretty terse.